algorithms/rotate_matrix.py

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+"""Python solution for CtCI 6th Edition Problem 1.7
+
+Given an image represented by an NxN matrix, where each pixel in the image is
+4 bytes, write a method to rotate the image by 90 degrees. Can you do this in
+place?
+
+Odd sized matrix
+>>> rotate_matrix([['q','w','e'],['a','s','d'],['z','x','c']])
```

Comment:
I would space these matrixes out because it is hard to look at these test cases visually and see that they are being rotated correctly